### PR TITLE
🚀 Dockerfile Python templates:  migration to Alpine Docker image

### LINKED ISSUE #5153: Refactoring of connector Dockerfile
Now most of AirByte python connectors are using  the Docker image `python:3.7-slim` without any optimization. And their final average sizes (with all libraries and software) are +-60MB.

I propose to use the following Dockerfile template:
```
FROM python:3.7.11-alpine3.14 as base
FROM base as builder
# build and load all requrements 

RUN apk --no-cache upgrade \
    && pip install --upgrade pip

WORKDIR /airbyte/integration_code
COPY setup.py ./
RUN pip install --prefix=/install .


FROM base
# copy all loaded and built libraries to a pure basic image
COPY --from=builder /install /usr/local

WORKDIR /airbyte/integration_code
# copy payload code only
COPY main.py ./
COPY <connector_name> ./<connector_name>


ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]

LABEL io.airbyte.version=<version>
LABEL io.airbyte.name=airbyte/<connector_name>
```
And as result the future size of docker images will be reduced several times

### PR DESCRIPTION
**Pull request by antixar**
## What
Updating of base Docker images for Python sources/destinations 

## Recommended reading order
1. `airbyte-integrations/connector-templates/destination-python/Dockerfile`
2. `airbyte-integrations/connector-templates/source-python/Dockerfile`
3. `airbyte-integrations/connector-templates/source-python-http-api/Dockerfile`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connector-templates/source-python/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-RUN apt-get update && apt-get install -y bash && rm -rf /var/lib/apt/lists/*
+# build and load all requirements 
+FROM base as builder
+WORKDIR /airbyte/integration_code
+
+# upgrade pip to the latest version
+RUN apk --no-cache upgrade \
+    && pip install --upgrade pip
 
+COPY setup.py ./
```
Comment (midavadim) [inline airbyte-integrations/connector-templates/source-python/Dockerfile:11]: Should we copy main.py as well?

Comment (antixar) [reply airbyte-integrations/connector-templates/source-python/Dockerfile:11]: for installation of all requirements we need a setup.py file only.
Sure we can add payload files too but they will not be used throughout this building stage.

**Review by bazarnov — APPROVED** [review]
Woks fine.

File: airbyte-integrations/connector-templates/destination-python/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+FROM python:3.7.11-alpine3.14 as base
 
-# Bash is installed for more convenient debugging.
-RUN apt-get update && apt-get install -y bash && rm -rf /var/lib/apt/lists/*
+# build and load all requirements 
+FROM base as builder
+WORKDIR /airbyte/integration_code
+
+# upgrade pip to the latest version
+RUN apk --no-cache upgrade \
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/destination-python/Dockerfile:8]: is bash included by default? could we install it if not?

**Review by sherifnada — COMMENTED** [review]
LGTM - can you make sure to follow the checklist to update the scaffolds and also to create an issue to do the same thing for java, and drop it in the GL java roadmap?

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connector-templates/destination-python/Dockerfile (hunk 2 above)
Comment (antixar) [reply airbyte-integrations/connector-templates/destination-python/Dockerfile:8]: Bash is not default Apline utility. I've added it. this is +1Mb for final size only)

**Comment by antixar** [comment]
> LGTM - can you make sure to follow the checklist to update the scaffolds and also to create an issue to do the same thing for java, and drop it in the GL java roadmap?

@sherifnada , Now we are gonna use these templates for all new connectors and update Dockerfile files for exists ones during refactoring and bug fixings. Later we will be able to update it for all sources. Now I have a concern about using this foreign basic image: python:3.7.11-alpine3.14. How will we update them in the future? Maybe will we create airbyte version of this image? I mean airbyte/base-python-alpine:latest. And we will be able to support it separately (updating version of Python or some default packages etc). And thus all new docker builds will use our new basic image automatically.
About Java builds I'll discuss about this with our Java team because I don't know all its nuances.

**Comment by sherifnada** [comment]
don't worry about base-python image, this is deprecated and we should actually move away from it to a normal python alpine image using the instructions here: https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/docs/legacy_cdk_migration.md

For Java, you don't need to do anything big: just create a github issue that says "do the same thing for java connectors" and leave it up to the java team 😜

**Pull request merged**